packages/react/src/shared.tsx

```diff
@@ -218,9 +247,8 @@ export function createSharedDependencies(
     transactionFactory,
     transactionGateway,
     transactionQueue,
-    walletFactory,
-    walletGateway,
-    providerFactory,
+    walletLogin,
+    walletLogout,
```

Comment:
I needed `walletLogout` in shared to init `CredentialsExpiryController` so I decided to do the same with `walletLogout`, then we don't need to expose `walletFactory` nor `walletGateway`.
I dunno if that's fine but I just thought to try to clean up it a bit. WDYT?